### PR TITLE
TailCutsDataVolumeReducer now instantiates TailCutsImageCleaner, fixes #1399

### LINKED ISSUE #1399: Abstract DataVolumeReducer instantiates TailCutsImageCleaner
In it's `__init__` method, the `DataVolumeReducer` instantiates a `TailCutsImageCleaner`, which should belong only to the `TailCutsDataVolumeReducer`, not the base class.

### PR DESCRIPTION
**Pull request by Hckjs**
#1399

### CODE REVIEW

File: ctapipe/image/reducer.py [hunk 1]
```diff
@@ -131,6 +118,36 @@ class TailCutsDataVolumeReducer(DataVolumeReducer):
         "normal TailcutCleaning is used.",
     ).tag(config=True)
 
+    def __init__(
```
Comment (MaxNoe) [inline ctapipe/image/reducer.py:121]: This should have the same signature as the base class to make it configurable from outside the of scripts through the config system.

So I'd suggest going the same route as with the gain selector. So create a traitlet `image_extractor_type` and then use `ImageExtractor.from_name` to create it in the `__init__`

Comment (Hckjs) [reply ctapipe/image/reducer.py]: The image_extractor type is now not shown in the docs for `TailCutsDataVolumeReducer`, is it normal? Don't know why.

Comment (MaxNoe) [reply ctapipe/image/reducer.py]: What do you mean with "in the docs"?

Comment (Hckjs) [reply ctapipe/image/reducer.py]: in the docs/_build/html/api after make docs

Comment (MaxNoe) [reply ctapipe/image/reducer.py]: You need to add a docstring to the class with a section `Attributes` for that.

**Comment by kosack** [comment]
I see a potential problem with using TailCutsImageCleaner - in fact it's a problem with traitlets.config, not with your code.  The configuration system only allows configuration by class, not by instance, so that means the TailCutsImageCleaner you use in this step would have to have the same configuration as the one used to clean the final images (assuming both the R1→DL0 and DL0→DL1 transforms are done in the same tool).  That could be an issue. One way to solve it would be just to create a subclass, like:

```python3
class DVRTailCutsImageCleaner(TailcutsImageCleaner):
    pass
```

then the cuts would be configured in a section called "DVRTailCutsImageCleaner", while "TailcutsImageCleaner" would be used for the DL1 cleaning. 

Or just use the `tailcuts_clean()` low level function internally, and make the thresholds explicitly a configuration of TailCutsDataVolumeReducer (similar to what was done in TwoPassImageCleaner)

**Comment by MaxNoe** [comment]
@kosack

That is not true. 

This will work perfectly fine because of the hierarchy. 

In the dl1 tool, you would configure:

`CameraCalibrator.TailCutsVolumeReducer.TailCutsImageCleaner`

for this one and

`TailCutsImageCleaner`,

Assuming that CameraCalibrator and the normal TailcutsImageCleaner afe direct children of the tool, like it is now in the stage1 process.

**Comment by kosack** [comment]
> In the dl1 tool, you would configure:
> CameraCalibrator.TailCutsVolumeReducer.TailCutsImageCleaner
> for this one and

I wasn't aware that that works in config files.  If so then great!  you can ignore my comment.

**Comment by MaxNoe** [comment]
Yes, json would look like this:

```json
{
    "CameraCalibrator": {
        "data_volume_reducer_type": "TailCutsDataVolumeReducer",
        "TailCutsDataVolumeReducer": {
            "TailCutsImageCleaner": {
                "picture_threshold": 10,
            }
        }
    }
}
```

**Review by MaxNoe — APPROVED** [review]

**Review by MaxNoe — CHANGES_REQUESTED** [review]
You should change this so you have a `image_extractor_type` traitlet so it has the same signature as the base class for `__init__`.

**Review by kosack — APPROVED** [review]

File: ctapipe/image/reducer.py [hunk 2]
```diff
@@ -131,6 +131,28 @@ class TailCutsDataVolumeReducer(DataVolumeReducer):
...
+        kwargs
+        """
+        super().__init__(config=config, parent=parent, subarray=subarray, **kwargs)
+
+        self.cleaner = TailcutsImageCleaner(parent=self, subarray=self.subarray)
+
+        self.image_extractor = ImageExtractor.from_name(
+            self.image_extractor_type,
+            subarray=self.subarray,
+            config=self.config
```
Comment (MaxNoe) [inline ctapipe/image/reducer.py:153]: this should be `parent=self`, not `config=self.config` to correctly build the hierarchy.

Here you would give the config for the DataVolumeReducer as the config for the ImageExtractor, which is wrong.

Comment (MaxNoe) [reply ctapipe/image/reducer.py:153]: Just as above with the cleaner

**Review by MaxNoe — APPROVED** [review]

File: ctapipe/image/reducer.py (hunk 2 above)
Comment (kosack) [reply ctapipe/image/reducer.py]: wouldn't `parent=self`  take care of this? Usually you do not have to pass the config manually

**Review by kosack — APPROVED** [review]

**Pull request merged**